### PR TITLE
Tembo OLAP Stack Benchmark

### PR DESCRIPTION
**Pull request by shhnwz**
Thank you for the contribution!

Please make sure that you've updated `index.html`. In order to do this run `generate-results.sh` from the root directory of ClickBench repository.

### CODE REVIEW

**Comment by shhnwz** [comment]
Hi @alexey-milovidov @nickitat ,
Opened a Pull request for Tembo Olap. Please review as per your convenience.

**Comment by samay-sharma** [comment]
Hi @alexey-milovidov , @nickitat : Could we get a review on this PR? We would like to get the results for our benchmark on the clickbench page.

**Comment by alexey-milovidov** [comment]
Yes, it is waiting for review.

File: tembo-olap/README.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
+# Instantiation
+
+Select OLAP Stack from https://cloud.tembo.io
+
+## Choose System Configuration:
+
+**CPU**: 16vCPU
```
Comment (chhetripradeep) [inline tembo-olap/README.md:7]: I was unable to find any instance with 16 vCPU in tembo cloud.

File: tembo-olap/README.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+gzip -d hits.tsv.gz
+```
+
+**Data** loading
+
+```
+psql postgresql://postgres:$PGPASSWORD@$PGHOST:5432 -t -c 'CREATE DATABASE test'
+
+psql postgresql://postgres:$PGPASSWORD@$PGHOST:5432/test -t < create.sql
```
Comment (chhetripradeep) [inline tembo-olap/README.md:46]: This fails with 
```
❯ psql postgresql://postgres:$PGPASSWORD@$PGHOST:5432/test -t < create.sql
ERROR:  access method "columnar" does not exist
```

**Review by chhetripradeep — CHANGES_REQUESTED** [review]
Will try it again once i get clarifications on the comments.

File: tembo-olap/run.sh [hunk 3]
```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+
+TRIES=3
+
+cat queries.sql | while read query; do
+    echo "$query";
```
Comment (chhetripradeep) [inline tembo-olap/run.sh:6]: Like in other benchmarks, would recommend to drop cache after every query in order to make sure that query isn't being served from linux page cache.

File: tembo-olap/README.md (hunk 2 above)
Comment (tinybit) [reply tembo-olap/README.md:46]: Same result. My instance:

```
https://cloud.tembo.io/orgs/org_2cpZgXjK2Ke7YOKb8T91DVEeTi5/clusters/inst_1708811361385_GzxAkc_8/home
```

File: tembo-olap/README.md (hunk 1 above)
Comment (shhnwz) [reply tembo-olap/README.md:14]: Hi @chhetripradeep ,
Thanks for looking into this pull request. Try now, earlier it wasn't enabled in the public domain. Added some more information regarding Instance type in the README.md file. Apologies for the inconvenience.

File: tembo-olap/run.sh (hunk 3 above)
Comment (shhnwz) [reply tembo-olap/run.sh:6]: Hi @chhetripradeep ,

Comment (chhetripradeep) [reply tembo-olap/run.sh:6]: Sounds good.

File: tembo-olap/README.md (hunk 2 above)
Comment (shhnwz) [reply tembo-olap/README.md:56]: Hi @chhetripradeep and @tinybit ,
The instance mentioned by @tinybit is also OLAP stack which is the correct one. I was under the assumption that instance stack was wrong..whereas it wasn't the case. Real bad guy was my create script,  I removed couple of DDL from the top leads to this regression. Now it's fixed.. It show work fine. 
Thanks Guys for the good catch.

Comment (tinybit) [reply tembo-olap/README.md:56]: Thanks for the quick update! We will continue with the review and hopefully finish it today

**Comment by tinybit** [comment]
@shhnwz I'm not quite sure I understand what is happening to my test DB on Tembo, but for some reason there's load and it's definitely not me. I'm not even connected at the moment and credentials to DB were not shared:
https://cloud.tembo.io/orgs/org_2cpZgXjK2Ke7YOKb8T91DVEeTi5/clusters/inst_1708811361385_GzxAkc_8/home

DB Usage metrics:
Screenshot 2024-02-27 at 11 37 58 AM

**Comment by tinybit** [comment]
@shhnwz Also, the DB still says it's "Configuring"
Screenshot 2024-02-27 at 11 42 52 AM
All I've done yesterday — disabled and then enabled "columnar storage" plugin. DB was in "Configuring state" since then

**Comment by tinybit** [comment]
@shhnwz For some reason I can't even delete the database:
Screenshot 2024-02-27 at 11 45 58 AM
Please help with this and previously mentioned issues.

**Comment by shhnwz** [comment]
> there's

Hi @tinybit ,
It's tembo's metrics collection and heartbeat queries, it's periodic but not very exhaustive.

**Comment by shhnwz** [comment]
> @shhnwz Also, the DB still says it's "Configuring"
> Screenshot 2024-02-27 at 11 42 52 AM
> All I've done yesterday — disabled and then enabled "columnar storage" plugin. DB was in "Configuring state" since then

Thanks @tinybit ,
This is known UI issue, things are working fine in the background. You can Ignore this system and create new system for benchmark. I can ask our support team to clean-up from their end.. Is that OK with you?

**Comment by tinybit** [comment]
> > @shhnwz Also, the DB still says it's "Configuring"

> Thanks @tinybit , This is known UI issue, things are working fine in the background. You can Ignore this system and create new system for benchmark. I can ask our support team to clean-up from their end.. Is that OK with you?

Yeah, please do clean-up, I'll create a new system. Thank you.

**Comment by samay-sharma** [comment]
> Also, 1M reads per hour, 277 reads p/s seems a bit excessive on an empty DB (from end-user perspective). It's irrelevant to the benchmarking topic, just for your consideration.

Absolutely @tinybit . We have an issue for this internally and will be fixing it soon.

**Comment by tinybit** [comment]
Also, please provide the ability to update your fork, I'd like to push my update that verified your results and cleans up the bench code.

```
$ git push
ERROR: Permission to tembo-io/ClickBench.git denied to tinybit.
```

**Comment by shhnwz** [comment]
> Also, please provide the ability to update your fork, I'd like to push my update that verified your results and cleans up the bench code.
> 
> ```
> $ git push
> ERROR: Permission to tembo-io/ClickBench.git denied to tinybit.
> ```

Hi @tinybit ,
Please check your email for the invitation.
https://github.com/tembo-io/ClickBench/invitations

**Comment by tinybit** [comment]
@shhnwz Please take a look at my changes. I've removed everything directly related to the way your instances are created, as this instruction will become obsolete over time, as your project moves forward and changes UI, etc.

**Comment by shhnwz** [comment]
> @shhnwz Please take a look at my changes. I've removed everything directly related to the way your instances are created, as this instruction will become obsolete over time, as your project moves forward and changes UI, etc.
> 

Thanks @tinybit ,
Looks Good to Me 😊

**Pull request merged**